python_modules/dagster/dagster/_core/definitions/sensor_definition.py

```diff
@@ -636,6 +637,201 @@ def assets_defs_by_key(self) -> Mapping[AssetKey, AssetsDefinition]:
         return self._assets_by_key
 
 
+@experimental
+class AssetSLASensorEvaluationContext(SensorEvaluationContext):
+    """The context object available as the argument to the evaluation function of a :py:class:`dagster.AssetSLASensorDefinition`."""
+
+    def __init__(
+        self,
+        instance_ref: Optional[InstanceRef],
+        last_completion_time: Optional[float],
+        last_run_key: Optional[str],
+        cursor: Optional[str],
+        repository_name: Optional[str],
+        repository_def: "RepositoryDefinition",
+        asset_keys: Optional[Sequence[AssetKey]] = None,
+        instance: Optional[DagsterInstance] = None,
+    ):
+        from dagster._core.selector.subset_selector import generate_asset_dep_graph
+
+        self._repository_def = repository_def
+
+        self._upstream_assets = generate_asset_dep_graph(
+            self._repository_def._assets_defs_by_key.values(),
+            self._repository_def.source_assets_by_key.values(),
+        )["upstream"]
+
+        self._assets_by_key: Dict[AssetKey, AssetsDefinition] = {}
+        # if no asset_keys supplied, default to all
+        for asset_key in asset_keys or repository_def._assets_defs_by_key.keys():
+            assets_def = self._repository_def._assets_defs_by_key.get(asset_key)
+            if assets_def is None:
+                raise DagsterInvalidDefinitionError(
+                    f"No asset with {asset_key} found in repository"
+                )
+            if not assets_def.slas_by_key.get(asset_key):
+                continue
+            self._assets_by_key[asset_key] = assets_def
+
+        self._next_cursor = cursor
```

Comment:
I think this should potentially work a bit more like the MultiAssetSensor stuff, where the user has to manually update the cursor, using a dictionary of whatever statuses they used?

In my mind, the main use case for this will be to only alert when a status changes from the last tick (most often when it goes from passing to failing, but sometimes you want the other direction as well). This is because it would be extremely annoying to get (e.g.) a slack message every 30 seconds whenever an SLA was failing.

This use case necessitates a cursor to keep track of the previous status